File: public/dsa/searching/next_greater_element_in_sorted_array/naive/next_greater_element_in_sorted_array.py
from bisect import bisect_right
from typing import List


class Solution:
    def getNextGreaterElement(self, arr: List[int], key: int) -> int:
        n = len(arr)
        if arr[n-1] <= key:
            return key
        for i in range(n):
            if arr[i] > key:
                return arr[i]


# tests-start
arr = [1, 2, 3, 3, 4, 4, 8, 10]
key = 4
sol = Solution()
print(sol.getNextGreaterElement(arr, key))
# tests-end

'''output
8
'''

File: public/dsa/searching/next_greater_element_in_sorted_array/naive/next_greater_element_in_sorted_array.cpp
#include <bits/stdc++.h>
using namespace std;

int getNextGreaterElement(vector<int> &arr, int key) {
    int n = arr.size();
    if (arr[n - 1] <= key) {
        return key;
    }
    for (int i = 0; i < n; i++) {
        if (arr[i] > key) {
            return arr[i];
        }
    }
}

// tests-start
int main() {
    vector<int> arr = {1, 2, 3, 3, 4, 4, 8, 10};
    int key = 4;
    cout << getNextGreaterElement(arr, key) << endl;
    return 0;
}
// tests-end

/*output
8
*/
